src/app/core/rpc/encryptionstatus.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable, Observer } from 'rxjs'; // use this for testing atm
+
+import { RPCService } from './rpc.service';
+
+@Injectable()
+export class EncryptionStatusService {
+
+  private _encryptionStatus: Observable<string>;
+  private _observerEncryptionStatus: Observer<string>;
+
+  private _encryptionStatusState: string = 'Locked';
+
+  constructor(private _rpc: RPCService) {
+
+    // register updates to encryptionStatus
+    this._rpc.register(this, 'getwalletinfo', null, this.setEncryptionStatus, 'time');
+
+    this._encryptionStatus = Observable.create(observer => this._observerEncryptionStatus = observer).publishReplay(1).refCount();
+    this._encryptionStatus.subscribe().unsubscribe();
+  }
+
+  private setEncryptionStatus(json: Object): void {
```

Comment:
no, it is descriptive of the serialization, but its currently deserialized to an Object..